catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/ProjectUploadRatingDialogTest.java

```diff
@@ -0,0 +1,172 @@
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.defaultprojectcreators.DefaultProjectCreator;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.SingleSprite;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.io.asynctask.ProjectSaveTask;
+import org.catrobat.catroid.testsuites.annotations.Cat;
+import org.catrobat.catroid.testsuites.annotations.Level;
+import org.catrobat.catroid.ui.ProjectUploadActivity;
+import org.catrobat.catroid.ui.controller.ProjectUploadController;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityTestRule;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.common.Constants.UPLOAD_RESULT_RECEIVER_RESULT_CODE;
+import static org.catrobat.catroid.ui.ProjectUploadActivity.NUMBER_OF_UPLOADED_PROJECTS;
+import static org.catrobat.catroid.ui.ProjectUploadActivity.PROJECT_DIR;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class ProjectUploadRatingDialogTest {
+
+	private ProjectUploadController projectUploadController;
+	private Bundle bundle;
+	private SharedPreferences sharedPreferences;
+	private static final String PROJECT_NAME = "projectName";
+
+	@Rule
+	public BaseActivityTestRule<ProjectUploadTestActivity> activityTestRule =
+			new BaseActivityTestRule<>(ProjectUploadTestActivity.class, false, false);
+
+	@Before
+	public void setUp() throws Exception {
+		Context context = InstrumentationRegistry.getTargetContext();
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		bundle = new Bundle();
+
+		Project project = new DefaultProjectCreator().createDefaultProject(PROJECT_NAME,
```

Comment:
please dont create a default project since this is awefully slow and overkill for what you need.
an empty project with a single brick or sprite in it would be sufficient.